src/client/Client.js

```diff
@@ -358,6 +359,20 @@ class Client extends BaseClient {
       .then(app => new ClientApplication(this, app));
   }
 
+  /**
+   * Obtains the widget from a guild, available for all guilds
+   * @param {GuildResolvable} guild The guild to fetch the widget for
+   * @returns {Promise<Widget>}
+   */
+  fetchWidget(guild) {
+    const id = this.guilds.resolveID(guild);
+    if (!id) return Promise.reject(new TypeError('INVALID_TYPE', 'guild', 'GuildResolvable'));
+    return this.api
+      .guilds(id)
+      ['widget.json'].get()
+      .then(data => new Widget(this, data, id));
```

Comment:
`data` comes with an `id`, no need to pass it through

Reply:
Not everytime, when the widget is desativate he don't comes with an id